packages/embark/src/lib/modules/parity/blockchain.js

```diff
@@ -0,0 +1,470 @@
+const async = require('async');
+const {spawn, exec} = require('child_process');
+const path = require('path');
+const constants = require('embark-core/constants');
+const ParityClient = require('./parityClient.js');
+// const ParityClient = require('./parityClient.js');
+// import { Proxy } from './proxy';
+import {IPC} from 'embark-core';
+
+import {compact, dappPath, defaultHost, dockerHostSwap, embarkPath, AccountParser} from 'embark-utils';
+const Logger = require('embark-logger');
+
+// time between IPC connection attempts (in ms)
+const IPC_CONNECT_INTERVAL = 2000;
+
+/*eslint complexity: ["error", 50]*/
+var Blockchain = function (userConfig, clientClass) {
+  this.userConfig = userConfig;
+  this.env = userConfig.env || 'development';
+  this.isDev = userConfig.isDev;
+  this.onReadyCallback = userConfig.onReadyCallback || (() => {});
+  this.onExitCallback = userConfig.onExitCallback;
+  this.logger = userConfig.logger || new Logger({logLevel: 'debug', context: constants.contexts.blockchain}); // do not pass in events as we don't want any log events emitted
+  this.events = userConfig.events;
+  this.proxyIpc = null;
+  this.isStandalone = userConfig.isStandalone;
+  this.certOptions = userConfig.certOptions;
+
+
+  let defaultWsApi = clientClass.DEFAULTS.WS_API;
+  if (this.isDev) defaultWsApi = clientClass.DEFAULTS.DEV_WS_API;
+
+  this.config = {
+    silent: this.userConfig.silent,
+    client: this.userConfig.client,
+    ethereumClientBin: this.userConfig.ethereumClientBin || this.userConfig.client,
+    networkType: this.userConfig.networkType || clientClass.DEFAULTS.NETWORK_TYPE,
+    networkId: this.userConfig.networkId || clientClass.DEFAULTS.NETWORK_ID,
+    genesisBlock: this.userConfig.genesisBlock || false,
+    datadir: this.userConfig.datadir,
+    mineWhenNeeded: this.userConfig.mineWhenNeeded || false,
+    rpcHost: dockerHostSwap(this.userConfig.rpcHost) || defaultHost,
+    rpcPort: this.userConfig.rpcPort || 8545,
+    rpcCorsDomain: this.userConfig.rpcCorsDomain || false,
+    rpcApi: this.userConfig.rpcApi || clientClass.DEFAULTS.RPC_API,
+    port: this.userConfig.port || 30303,
+    nodiscover: this.userConfig.nodiscover || false,
+    mine: this.userConfig.mine || false,
+    account: {},
+    whisper: (this.userConfig.whisper !== false),
+    maxpeers: ((this.userConfig.maxpeers === 0) ? 0 : (this.userConfig.maxpeers || 25)),
+    bootnodes: this.userConfig.bootnodes || "",
+    wsRPC: (this.userConfig.wsRPC !== false),
+    wsHost: dockerHostSwap(this.userConfig.wsHost) || defaultHost,
+    wsPort: this.userConfig.wsPort || 8546,
+    wsOrigins: this.userConfig.wsOrigins || false,
+    wsApi: this.userConfig.wsApi || defaultWsApi,
+    vmdebug: this.userConfig.vmdebug || false,
+    targetGasLimit: this.userConfig.targetGasLimit || false,
+    syncMode: this.userConfig.syncMode || this.userConfig.syncmode,
+    verbosity: this.userConfig.verbosity,
+    proxy: this.userConfig.proxy
+  };
```

Comment:
Those are the same as Geth no? Anyway we could have a shared `blockchain` module that shares this config?

Reply:
See my comment below.